tests/test_clpir.py

```diff
@@ -744,5 +762,101 @@ def test_log_event_match(self) -> None:
         self.assertEqual(log_event.match_query(query), True, description)
 
 
+class TestCaseDecoderBuffer(unittest.TestCase):
+    """
+    Class for testing clp_ffi_py.ir.DecoderBuffer.
+    """
+
+    input_src_dir = "DecoderBufferTestSrc"
+
+    def test_buffer_protocol(self) -> None:
+        """
+        Tests whether the buffer protocol is disabled by default.
+        """
+        byte_array: bytearray = bytearray(b"Hello, world!")
+        byte_stream: io.BytesIO = io.BytesIO(byte_array)
+        decoder_buffer: DecoderBuffer = DecoderBuffer(byte_stream)
+        exception_captured: bool = False
+        try:
+            byte_stream.readinto(decoder_buffer)  # type: ignore
+        except TypeError:
+            exception_captured = True
+        self.assertTrue(
+            exception_captured, "The buffer protocol should not be enabled in Python layer."
+        )
+
+    def test_streaming_small_buffer(self) -> None:
+        """
+        Tests DecoderBuffer's functionality using the small buffer capacity.
+        """
+        buffer_capacity: int = 1024
+        self.__launch_test(buffer_capacity)
+
+    def test_streaming_default_buffer(self) -> None:
+        """
+        Tests DecoderBuffer's functionality using the default buffer capacity.
+        """
+        self.__launch_test(None)
+
+    def test_streaming_large_buffer(self) -> None:
+        """
+        Tests DecoderBuffer's functionality using the large buffer capacity.
+        """
+        buffer_capacity: int = 16384
+        self.__launch_test(buffer_capacity)
+
+    def __launch_test(self, buffer_capacity: Optional[int]) -> None:
+        """
+        Tests the DecoderBuffer by streaming the files inside `test_src_dir`.
+
+        :param self
+        :param buffer_capacity: The buffer capacity used to initialize the
+            decoder buffer.
+        """
+        current_dir: Path = Path(__file__).resolve().parent
+        test_src_dir: Path = current_dir / TestCaseDecoderBuffer.input_src_dir
+        for file_path in test_src_dir.rglob("*"):
+            if not file_path.is_file():
+                continue
+            streaming_result: bytearray
+            decoder_buffer: DecoderBuffer
+            random_seed: int = floor(time.time() * 1000) % 1000
```

Comment:
You might as well generate a "proper" seed: https://docs.python.org/3/library/random.html

Reply:
Thanks. It's good to know that python `random` methods will use the system time as the seed by default (I thought it is the same as C/C++ that defaults to 0)